comment/validators.py

```diff
@@ -0,0 +1,101 @@
+        """
+        if self.api:
+            return super().dispatch(request, *args, **kwargs)
+        try:
+            self.validate(request)
+        except CommentBadRequest as exc:
+            return JsonResponse({'type': _('error'), 'detail': _(exc.detail)}, status=400)
+        return super().dispatch(request, *args, **kwargs)
+
+    def validate(self, request):
+        self.app_name = request.GET.get("app_name") or request.POST.get("app_name")
+        self.model_name = request.GET.get("model_name") or request.POST.get("model_name")
+        self.model_id = request.GET.get("model_id") or request.POST.get("model_id")
+        self.parent_id = request.GET.get("parent_id") or request.POST.get("parent_id")
+
+
+class ContentTypeValidator(BaseValidatorMixin):
+    def validate(self, request):
+        super().validate(request)
+        if not self.model_name:
+            self.error = "model name must be provided"
+            raise CommentBadRequest(self.error)
+        if not self.model_id:
+            self.error = "model id must be provided"
+            raise CommentBadRequest(self.error)
+        if not self.app_name:
+            self.error = 'app name must be provided'
+            raise CommentBadRequest(self.error)
+
+        if not ContentType.objects.filter(app_label=self.app_name).exists():
+            self.error = f'{self.app_name} is NOT a valid app name'
+            raise CommentBadRequest(self.error)
+        try:
+            model_name = self.model_name.lower()
+            ct = ContentType.objects.get(model=model_name).model_class()
+            model_class = ct.objects.filter(id=self.model_id)
+            if not model_class.exists() and model_class.count() != 1:
+                self.error = f'{self.model_id} is NOT a valid model id for the model {self.model_name}'
+                raise CommentBadRequest(self.error)
+        except ContentType.DoesNotExist:
+            self.error = f'{self.model_name} is NOT a valid model name'
+            raise CommentBadRequest(self.error)
+        except ValueError:
+            self.error = f'model id must be an integer, {self.model_id} is NOT'
+            raise CommentBadRequest(self.error)
+
+
+class ParentIdValidator(BaseValidatorMixin):
+    def validate(self, request):
+        super().validate(request)
+        if not self.parent_id or self.parent_id == '0':
+            return
+        try:
+            Comment.objects.get(id=self.parent_id, object_id=self.model_id)
+        except Comment.DoesNotExist:
+            self.error = (
+                f'{self.parent_id} is NOT a valid id for a parent comment or '
+                'the parent comment does NOT belong to the provided model object'
+            )
+            raise CommentBadRequest(self.error)
+        except ValueError:
+            self.error = f'the parent id must be an integer, {self.parent_id} is NOT'
+            raise CommentBadRequest(self.error)
```

Comment:
Code readability could be increased if we could refactor this to

 ```python

    def _clean_parent_id(self, val):
        if not val or val == '0':
            val = None
        return val

    def _set_error_for_parent_id(self, val):
        if val:
            model_id = self.data.get('model_id')
            try:
                if not Comment.objects.filter(id=val, object_id=model_id).exists():
                    self.error = _(
                        f'{val} is NOT a valid id for a parent comment or '
                        'the parent comment does NOT belong to this model object'
                        )
                    raise CommentBadRequest(self.error)
            except ValueError:
                self.error = f"the parent id must be an integer, {val} is NOT"
                raise CommentBadRequest(self.error)

    def validate(self, request):
         super().validate(request)
         self.parent_id = self._clean_parent_id(self.parent_id)
         self._set_error_for_parent_id( self.parent_id)

```

Reply:
I believe the old one is better. 3 indentation levels make the readability less 😄

Reply:
for what it's worth, it also violates [single responsibility principle](https://en.m.wikipedia.org/wiki/Single-responsibility_principle)